scripts/dumps/bulk_genome_dump_from_core.py

```diff
@@ -0,0 +1,343 @@
+        shell (bool): If True, the specified command will be executed through the shell.
+        use_job_scheduler (bool): If True, the command will be submitted to a job scheduler.
+
+    Returns:
+        str: The subprocess output or None otherwise.
+    """
+    if use_job_scheduler:
+        job_scheduler = detect_job_scheduler()
+
+        if job_scheduler == "NONE":
+            logging.error("No job scheduler detected.")
+            sys.exit(1)
+
+        if job_scheduler == "SLURM":
+            command = [
+                "sbatch",
+                "--time=1-00",
+                "--mem-per-cpu=4gb",
+                "--cpus-per-task=1",
+                "--export=ALL",
+                f"--output={stdout_file}",
+                f"--error={stderr_file}",
+                f"--job-name={job_name}",
+                f"--wrap={' '.join(command)}",
+            ]
+        elif job_scheduler == "LSF":
+            command = [
+                "bsub",
+                "-W",
+                "1:00",
+                "-R",
+                "rusage[mem=4096]",
+                "-J",
+                job_name,
+                f"-o {stdout_file}",
+                f"-e {stderr_file}",
+            ] + command
+
+    logging.info("Running: %s", " ".join(command))
+    with subprocess.Popen(
+        command,
+        shell=shell,
+        cwd=work_dir,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    ) as process:
+        output, stderr = process.communicate()
+
+        if process.returncode != 0:
+            out = f"stdout={output}, stderr={stderr}"
+            logging.error(
+                "Command %s exited %d: %s", " ".join(command), process.returncode, out
+            )
+            raise RuntimeError(
+                f"Command {' '.join(command)} exited {process.returncode}: {out}"
+            )
+
+        logging.info("Successfully ran: %s", " ".join(command))
+        return output.strip()
+
+
+def download_file(
+    host,
+    port,
+    core_db,
+    fasta_filename,
+    stdout_file="/dev/null",
+    stderr_file="/dev/null",
+    genome_component="",
+    mask="soft",
+):
+    """
+    Download the FASTA file from the core DB using a PERL script `dump_genome_from_core.pl`.
+
+    Args:
+        host (str): The DB host name.
+        port (str): The port number.
+        core_db (str): The core_db name.
+        fasta_filename (str): The name given for the FASTA file.
+        mask (str): The mask format for the FASTA file.
+
+    Returns:
+        str: The output of the subprocess call to download the FASTA file.
+    """
+    try:
+        work_dir = os.path.join(
+            os.environ["ENSEMBL_ROOT_DIR"],
+            "ensembl-compara",
+            "scripts",
+            "dumps",
+        )
+        script = "dump_genome_from_core.pl"
+
+        perl_call = [
+            "perl",
+            os.path.join(work_dir, script),
```

Comment:
Could the directory containing the script be renamed to something like `script_dir`? The name `work_dir` gives the impression that intermediate files will be stored in it, rather than the script being called.
```suggestion
            os.path.join(script_dir, script),
```